### ISSUE TITLE
.git-ftp-ignore not respected inside .git-ftp-included directories

### ISSUE THREAD

**Issue by jtojnar** [git-ftp/git-ftp#301]
I use composer and have `.git-ftp-include` set up to upload the `vendor` directory whenever the lock file changes:

```
vendor/:composer.lock
```

I also have `.git-ftp-ignore` file containing the following:

```
.git
doc
docs
test
tests
```

However, I still see hundreds of lines like this: 

```
Buffered for upload 'vendor/ezyang/htmlpurifier/.git/objects/fd/b58c2d376cd9b09fc9dc1d564e9c34fb30060a'.
```

While completely unnecessary on the server, these files constitute the majority of the uploaded data. It would be nice it they could be ignored.

**Comment by efes0** [comment]
your `.git-ftp-ignore` is looking for a file called ".git" to ignore not a ".git" folder

see here: [ignoring-files-to-be-synced](https://github.com/git-ftp/git-ftp/blob/master/man/git-ftp.1.md#ignoring-files-to-be-synced)

> For example, ignoring everything in a directory named `config`:
> `config/*`

**Comment by mkllnk** [comment]
Or to be more specific:
```
vendor/*/*/.git/*
```

**Comment by efes0** [comment]
wouldn't `*/.git/*` match as well?

**Comment by mkllnk** [comment]
Yes, you are right. That would match any file in a .git folder in any subfolder.

**Comment by jtojnar** [comment]
> your `.git-ftp-ignore` is looking for a file called ".git" to ignore not a ".git" folder

I have tried both `.git` and `.git/*`. It did not occur to me I have to use `*/.git/*` since in `.gitignore` file adding `.git` would suffice.

Now, thinking about it little more, **mkllnk**’s solution would be preferable to avoid false matches (e.g. legitimate library directories named test). However, `vendor/*/*/.git/*` roughly corresponds to this regex `
^vendor/.*/.*/\.git/.*$`, while preventing false matches would require something like `^vendor/[^/]*/[^/]*/\.git/.*$`.

Is it possible to create pattern like that? Or would it be possible to support [`.gitignore`-compatible](https://git-scm.com/docs/gitignore#_pattern_format) matching?

**Comment by mkllnk** [comment]
The `.git-ftp-ignore` file supported regexes in the past. But that confused a lot of people. It would be better to become `.gitignore`-compatible. That would mean adding the support for a leading `/` and looking into subfolders otherwise. Would you be confident enough to implement that yourself? My time is very limited at the moment.

**Comment by jtojnar** [comment]
Unfortunately, I have rather busy schedule myself. Unless my procrastination leads that way, I will not be able to implement it any time soon.
